common/pkgfiles/referencedfilelist.cpp

```diff
@@ -541,23 +653,40 @@ static void dfuCopy(const IPropertyTree *directories, IDFUWorkUnit *publisherWu,
     if(isEmptyString(destPlane))
         throw makeStringException(-1, "Destination node group not specified.");
 
-    PROGLOG("Copy from [%s]  %s to %s", isEmptyString(srcDali) ? "local" : srcDali, sourceLogicalName, destLogicalName);
+    PROGLOG("Copy from %s[%s]  %s to %s", (!isEmptyString(srcLocation) && useRemoteStorage) ? "remote" : "", isEmptyString(srcLocation) ? "local" : srcLocation, sourceLogicalName, destLogicalName);
 
     StringBuffer destFolder, destTitle, defaultFolder, defaultReplicateFolder;
     DfuParseLogicalPath(directories, destLogicalName, destPlane, destFolder, destTitle, defaultFolder, defaultReplicateFolder);
 
     CDfsLogicalFileName logicalName;
     logicalName.set(sourceLogicalName);
-    if (!isEmptyString(srcDali))
+    if (!isEmptyString(srcLocation))
     {
-        SocketEndpoint ep(srcDali);
-        if (ep.isNull())
-            throw MakeStringException(-1, "ReferencedFile Copy %s: cannot resolve SourceDali network IP from %s.", sourceLogicalName, srcDali);
+        if (useRemoteStorage)
+        {
+            StringBuffer remoteSpec;
+            VStringBuffer lfn("remote::%s::", srcLocation);
+            if (logicalName.isRemote())
+                logicalName.getRemoteSpec(remoteSpec, lfn);
+            else
+            {
+                logicalName.clearForeign();
+                logicalName.get(lfn);
+            }
+            logicalName.set(lfn);
+            sourceLogicalName = logicalName.get();
+        }
+        else
+        {
+            SocketEndpoint ep(srcLocation);
+            if (ep.isNull())
+                throw MakeStringException(-1, "ReferencedFile Copy %s: cannot resolve SourceDali network IP from %s.", sourceLogicalName, srcLocation);
 
-        logicalName.setForeign(ep,false);
+            logicalName.setForeign(ep,false);
+        }
     }
-
-    Owned<IDistributedFile> file = queryDistributedFileDirectory().lookup(logicalName, user, AccessMode::tbdRead, false, false, nullptr, defaultPrivilegedUser);
+    //Pass forceAllowForeign=true because: 1. we need backward compatibility of configurations, 2. this may be the only place we want to allow foreiegn (to copy files in)
```

Comment:
Not sure about this approach, extending the interface with forceAllowForeign
There is already an option that can be configured per component (at least if it has an expert section, which could be extended to others if need be) or globally, isn't that sufficient?

Reply:
I'll remove the extra parameter.  This hit me at the last minute while rebasing.  At first I thought my use was an exception but I've come around to requiring the config to set the expert option.  If we are copying query and packagemaps from foreign locations we're going to want to be able to do similar via eclwatch for certain scenarios.

Reply:
should "expert" need to be added to the esp schema at this point, or can be done after in a separate JIRA?